### PR TITLE
Wiz 1.2.0 - multiple auth endpoints

### PR DESCRIPTION
**Pull request by solalraveh**
The commands shown as new in the diff aren't new, but are already from 1.1.0. 

This 1.2.0 release is to support different authentication endpoints in Wiz (legacy / modern, gov / commercial, etc).
Admins can now choose with auth endpoint to use for authenticating with Wiz, thus having the ability to support both Commercial and Gov deployments of Wiz, and legacy vs modern auth vendors of Wiz backend. One param to support them all.
Existing integrations (customers) won't be broken, since this integration de facto supported only Wiz Commercial in the code until today. The new auth endpoint param isn't marked as mandatory for now so we can support existing customers that will upgrade and will not populate this param. The code checks if it's empty and assigns a default value to represent the way the code was until today.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by solalraveh** [comment]
The commands shown as new in the diff aren't new, but are already from 1.1.0.

**Comment by solalraveh** [comment]
Adding a couple field changes and then verifying tests, then removing draft mode.
This 1.2.0 release is to support different authentication endpoints in Wiz (legacy / modern, gov / commercial, etc).

**Comment by solalraveh** [comment]
@kgal-pan , should be ready.
https://app.circleci.com/pipelines/github/demisto/content/185729/workflows/c0e39954-1694-45bd-9be8-170a59266738/jobs/559074?invite=true#step-108-195 This isn'e a breaking change (the only one failure here in the checks), since it was supposed to be a mandatory param from the get go, and the integration doesn't work without it either way. It's also documented as a mandatory one.

File: Packs/Wiz/ReleaseNotes/1_2_0.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#### Integrations
+##### Wiz
+- Added support for dual authenticaion mode. Since Wiz is shifting from our legacy authentication provider (Auth0) to Amazon Cognito, this integration now supports both authentication endpoints, we actually support any authentication endpoint now.
+    - The user now needs to configure the authentication endpoint of their tenant. The authentication endpoint is unique to the environment (e.g., commercial, gov) and authentication provider
+    - The integration assesses the authentication endpoint and based on that chooses the correct authentication payload to go with it
+    - Existing integrations that upgrade to 1.2.0 will need to add an authentication endpoint after the upgrade
```
Comment (kgal-pan) [inline Packs/Wiz/ReleaseNotes/1_2_0.md:6]:
```suggestion
- The user now needs to configure the authentication endpoint of their tenant. The authentication endpoint is unique to the environment (e.g., commercial, gov) and authentication provider
- The integration assesses the authentication endpoint and based on that chooses the correct authentication payload to go with it
- Existing integrations that upgrade to 1.2.0 will need to add an authentication endpoint after the upgrade
```

File: Packs/Wiz/ReleaseNotes/1_2_0.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+##### Wiz Mapper Webhook
+- Updated the mapper to populate the Resolution Recommendation field.
+
+#### Layouts
+##### Wiz Layout
+- Added the new field **Wiz Issue Resolution Recommendation** to the layout.
+
+#### Incident Fields
+- **Wiz Issue Resolution Recommendation**
+    - Added a new incide field, Wiz Issue Resolution Recommendation, that holds the suggested resolution steps from Wiz. Type: markdown.
```
Comment (kgal-pan) [inline Packs/Wiz/ReleaseNotes/1_2_0.md:21]:
```suggestion
- Added a new incide field, Wiz Issue Resolution Recommendation, that holds the suggested resolution steps from Wiz. Type: markdown.
```

**Comment by YuvHayun** [comment]
Hi @solalraveh, thanks for the contribution.
Except from @kgal-pan comments, I'd recommend re-updating release notes with the -bc and writing there about the added required field so we'll have it documented just in case.
Also, let's wait for @idovandijk to approve the incidentFields/classifiers/layouts changes.

File: Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Resolution_Recommendation.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "required": false,
+    "neverSetAsRequired": false,
+    "isReadOnly": false,
+    "useAsKpi": false,
+    "locked": false,
+    "system": false,
+    "content": true,
+    "group": 0,
+    "hidden": false,
+    "associatedToAll": true,
```
Comment (idovandijk) [inline Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Resolution_Recommendation.json:20]: Please associate the incident field to the relevant incident type.

**Review by idovandijk — CHANGES_REQUESTED** [review]
Thank you for the contribution.
I added one comment to fix.

File: Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Resolution_Recommendation.json (hunk 3 above)
Comment (solalraveh) [reply Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Resolution_Recommendation.json]: Done in new commit

**Comment by solalraveh** [comment]
@YuvHayun there is no breaking change for existing customers as we built a fallback mechanism to that field, that if a customer upgrades and forgets/neglects populating this field, then it defaults back to what we supported in the auth provider URL before 1.2.0.
New integrations need to populate it.
Are you still seeing a breaking change?

The other parameter (the one that marks the circleci checks as failing (api_endpoint) became mandatory. It always should have been as such, but it wasn't set to true. Any existing integration wouldn't have been able to set ip up while leaving this field empty due to the code checks. So you can say it's tech debt to mark the parameter as mandatory. There is no integration that works with it being empty either way and the logs/tests show in the UI show it :)
I hope it clarifies it.
@kgal-pan mentioned you can still force-push it since it's not really a bc

**Comment by YuvHayun** [comment]
> @YuvHayun there is no breaking change for existing customers as we built a fallback mechanism to that field, that if a customer upgrades and forgets/neglects populating this field, then it defaults back to what we supported in the auth provider URL before 1.2.0. New integrations need to populate it. Are you still seeing a breaking change?
> 
> The other parameter (the one that marks the circleci checks as failing (api_endpoint) became mandatory. It always should have been as such, but it wasn't set to true. Any existing integration wouldn't have been able to set ip up while leaving this field empty due to the code checks. So you can say it's tech debt to mark the parameter as mandatory. There is no integration that works with it being empty either way and the logs/tests show in the UI show it :) I hope it clarifies it. @kgal-pan mentioned you can still force-push it since it's not really a bc

Got it. I'll trigger the instance test as soon as the build is done.

**Comment by solalraveh** [comment]
> > @YuvHayun there is no breaking change for existing customers as we built a fallback mechanism to that field, that if a customer upgrades and forgets/neglects populating this field, then it defaults back to what we supported in the auth provider URL before 1.2.0. New integrations need to populate it. Are you still seeing a breaking change?
> > The other parameter (the one that marks the circleci checks as failing (api_endpoint) became mandatory. It always should have been as such, but it wasn't set to true. Any existing integration wouldn't have been able to set ip up while leaving this field empty due to the code checks. So you can say it's tech debt to mark the parameter as mandatory. There is no integration that works with it being empty either way and the logs/tests show in the UI show it :) I hope it clarifies it. @kgal-pan mentioned you can still force-push it since it's not really a bc
> 
> Got it. I'll trigger the instance test as soon as the build is done.

It was done again, but it needed an updated dockerimage again, so we'll wait once more :)

**Comment by solalraveh** [comment]
Only thing that still fails is this: https://app.circleci.com/pipelines/github/demisto/content/186804/workflows/9480101a-ef82-4a0d-843d-742fa7715380/jobs/561033?invite=true#step-108-10

About your slack integration? Either way, the coverage is about 90% and everything else is checked and tested (based on previous commits were we saw it was fine). 
Thanks!

**Comment by kgal-pan** [comment]
> Only thing that still fails is this: [app.circleci.com/pipelines/github/demisto/content/186804/workflows/9480101a-ef82-4a0d-843d-742fa7715380/jobs/561033?invite=true#step-108-10](https://app.circleci.com/pipelines/github/demisto/content/186804/workflows/9480101a-ef82-4a0d-843d-742fa7715380/jobs/561033?invite=true#step-108-10)
> 
> About your slack integration? Either way, the coverage is about 90% and everything else is checked and tested (based on previous commits were we saw it was fine). Thanks!

@solalraveh - Can you try to pull from upstream? I'm seeing [more recent builds](https://app.circleci.com/pipelines/github/demisto/content/186826/workflows/c8348658-a7ea-4d07-a511-5be8c100b92d/jobs/561086/steps?invite=true#step-108-8) run successfully.

**Comment by solalraveh** [comment]
> > Only thing that still fails is this: [app.circleci.com/pipelines/github/demisto/content/186804/workflows/9480101a-ef82-4a0d-843d-742fa7715380/jobs/561033?invite=true#step-108-10](https://app.circleci.com/pipelines/github/demisto/content/186804/workflows/9480101a-ef82-4a0d-843d-742fa7715380/jobs/561033?invite=true#step-108-10)
> > About your slack integration?
> 
> @solalraveh - Can you try to pull from upstream? I'm seeing [more recent builds](https://app.circleci.com/pipelines/github/demisto/content/186826/workflows/c8348658-a7ea-4d07-a511-5be8c100b92d/jobs/561086/steps?invite=true#step-108-8) run successfully.

I did that about an hour or two go.
image

Can do again.

**Comment by solalraveh** [comment]
Done again.

**Review by idovandijk — COMMENTED** [review]
Approved from my side. @YuvHayun / @kgal-pan up to you

**Comment by YuvHayun** [comment]
@solalraveh - seems like there're build problems..
It should be fixed in an hr or two. Please re-pull from master in about an hr or two and see if it's fixed.
sorry for the inconvenience!

**Comment by solalraveh** [comment]
> @solalraveh - seems like there're build problems.. It should be fixed in an hr or two. Please re-pull from master in about an hr or two and see if it's fixed. sorry for the inconvenience!

@YuvHayun I ran the command like yesterday again to pull from master, and again (not the first time), it pulled a lot of changes done by other integrations into this PR. 
How do we solve this (for this PR, so it only shows our changes) and stop it from happening again? 

For awareness, the command I ran today, is the same as yesterday that had no issues (and is the one quoted from the circleci logs) : `./Utils/git_pull_master_into_fork.sh `

**Comment by solalraveh** [comment]
@YuvHayun What I mentioned about other integrations showing up doesn't show anymore. Odd.
Either way, the circleci checks still fail on https://app.circleci.com/pipelines/github/demisto/content/187152/workflows/48adb876-0873-4a8e-90ad-7552c9d6c7fe/jobs/561719?invite=true#step-108-19
after the additional pull from master.

**Comment by kgal-pan** [comment]
> Regarding the build error, we're aware of it and seeing it in other PRs as well. The fix is in #20875. So once merged, a pull from `master` should unblock the build.

@solalraveh @ariel-wiz - Please sync with `master`.

**Comment by solalraveh** [comment]
> > Regarding the build error, we're aware of it and seeing it in other PRs as well. The fix is in #20875. So once merged, a pull from `master` should unblock the build.
> 
> @solalraveh @ariel-wiz - Please sync with `master`.

@kgal-pan done. Waiting for all pipelines to finish.

**Review by YuvHayun — APPROVED** [review]

**Review by idovandijk — APPROVED** [review]

**Comment by YuvHayun** [comment]
@solalraveh thank you very much for the contribution!
Merging this pr.

**Pull request merged**